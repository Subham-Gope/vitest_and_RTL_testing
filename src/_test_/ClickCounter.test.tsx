// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import ClickCounter from "../components/ClickCounter";

// describe("Testing the ClickCounter.tsx file", () => {
//   test("clicks the button to increase the counter value", async () => {
//     //Setting up the userEvent
//     const user = userEvent.setup();

//     const { container } = render(<ClickCounter message={"Hello"} />);

//     expect(container).toMatchSnapshot();

//     const counterText = screen.getByText(/counter value/i);
//     expect(counterText).toBeInTheDocument();
//     expect(counterText).toHaveTextContent("Counter Value: 0");

//     const button = screen.getByRole("button", { name: /click me/i });
//     expect(button).toBeInTheDocument();

//     // Initiating the user interactions
//     await user.click(button);
//     expect(counterText).toHaveTextContent("Counter Value: 1");
//   });
// });
