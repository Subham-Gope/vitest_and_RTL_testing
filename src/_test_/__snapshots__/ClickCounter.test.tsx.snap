// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the ClickCounter.tsx file > clicks the button to increase the counter value 1`] = `
<div>
  <h1>
    Hello
  </h1>
  <div>
    <h2
      class="mt-8 text-4xl "
    >
      Counter Value: 
       
      <span
        class="font-bold text-blue-600"
      >
        0
      </span>
    </h2>
    <button
      class="bg-orange-400 px-12 py-4 rounded-[16px] text-white font-bold outline-none mt-8"
    >
      Click Me
    </button>
  </div>
</div>
`;
